### PR TITLE
fix: workbench header icon alignments

### PR DESCRIPTION
**Pull request by mshaaban0**
# Purpose of PR

Align Workbench header icons properly
**Before**
Screenshot 2022-05-12 at 16 40 16

**After**
Screenshot 2022-05-12 at 16 39 52

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/button/src/Button/Button.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ function _Button<E extends React.ElementType = typeof BUTTON_DEFAULT_TAG>(
     <>
       {startIcon && iconContent(startIcon)}
       {children && (
-        <Box as="span" display="block" className={styles.buttonContent}>
```
Comment (bgutsol) [inline packages/components/button/src/Button/Button.tsx:85]: Is this fix also related to Workbench fix?

**Review by bgutsol — COMMENTED** [review]
Workbench fix looks good. But I would ask you to revert the Button change

File: packages/components/button/src/Button/Button.tsx (hunk 1 above)
Comment (mshaaban0) [reply packages/components/button/src/Button/Button.tsx:85]: @bgutsol Yes it is related. SVGs inside block elements adds extra space "browser behaviour" display `inline-flex` is one way to fix it

Comment (bgutsol) [reply packages/components/button/src/Button/Button.tsx:85]: But we don't pass SVGs as children to the button, we have `startIcon/endIcon` props for that. What exactly you are trying to fix with it?

Comment (mshaaban0) [reply packages/components/button/src/Button/Button.tsx]: Reverted: Back button in workbench was using button and injecting an icon as a children. Instead I refactored it to use IconButton which uses `flex` 🤷🏼

Comment (bgutsol) [reply packages/components/button/src/Button/Button.tsx]: Awesome, thank you 🙌

**Review by bgutsol — APPROVED** [review]
